### ISSUE TITLE
visual method

### ISSUE THREAD

**Issue by maomaocun** [magic-research/Dataset_Quantization#11]
What is the method that makes the visual of the feature of bins?T-Sne?

**Comment by vimar-gu** [comment]
Yes, we used t-SNE for the bin visualization. The sample indices of each bin are recorded and emphasized in different figures.

**Comment by maomaocun** [comment]
Can I get the specific source code? I tried both the dataset images and the features generated by the pre-trained model, but I found that neither was very obvious.
<img width="507" alt="image" src="https://github.com/user-attachments/assets/017ba108-e255-485a-9215-9b823e8ef78e">
<img width="523" alt="image" src="https://github.com/user-attachments/assets/288077f7-3c9a-4713-a944-04e37b827901">

**Comment by vimar-gu** [comment]
Please try visualizing samples from one single class, which is also the case presented in the paper. Running t-SNE on a large number of samples will decrease the quality of dimension reduction.

**Comment by maomaocun** [comment]
<img width="521" alt="image" src="https://github.com/user-attachments/assets/103c0720-7a56-4e03-945f-d9dc4dd1179c">
<img width="537" alt="image" src="https://github.com/user-attachments/assets/f41d9519-b147-42bf-9442-fbb53f52c813">
It looks a bit strange, but it seems to reflect the characteristics of the data set distribution. Do you have any guidance? I performed t-SNE operation on the gradient generated by the pre-trained model of the data set, and the final image was normalized. The t-SNE parameters are as follows
<img width="604" alt="image" src="https://github.com/user-attachments/assets/a41706ac-f427-4833-a9d8-5ca4d6e7df25">

**Comment by vimar-gu** [comment]
You should simply use the extracted features to perform t-SNE. The figures you presented seem kind of strange. There seem to be two groups of features. But for bin selection, there only exist real images, so there should not be such distribution gap. 

I'm sorry I don't have the original script now. But the general procedure should be:
- Selecting bins and obtain the indices for each bin.
- Conducting t-SNE on the extracted features of one class.
- Display the original distribution and selected samples with different colors.

Do make sure that the sample order is consistent when performing bin selection and figure drawing.

**Comment by maomaocun** [comment]
<img width="543" alt="image" src="https://github.com/user-attachments/assets/3b51c419-87a6-4904-b1c9-e772b554a9ec">
<img width="552" alt="image" src="https://github.com/user-attachments/assets/3199512d-cf7e-40d2-8d43-276a33e0bc1e">
bin
The first picture is bin1, and the second picture is bin10. It seems to be a little closer to the meaning of your paper.

**Comment by vimar-gu** [comment]
Thanks for the feedback. But the figures are still not correct. 

At the bin selection stage, onyl original images are selected. It means that the red dots (selected samples) should exactly cover the corresponding blue dots (original samples). And the red dots in the first figure should be of blue color in the other figures. But the presented two figures don't meet the principle. The features of a class should be processed through t-SNE all together. Then only the selected indices are assigned with a different color. Please check the whole process again and see if the red dots are based on different features.
